python_modules/dagster/dagster/_utils/schedules.py

```diff
@@ -379,12 +378,185 @@ def _find_schedule_time(
         raise Exception(f"Unexpected schedule type {schedule_type}")
 
 
+def _find_next_croniter_time(
+    cron_string, timezone_str: str, start_timestamp: float, ascending: bool
+):
+    """Use croniter to determine the next timestamp matching the passed in cron string
+    that is past the passed in UTC timestamp. croniter can only be trusted to compute
+    non-timezone aware cron intervals, so we first figure out the time corresponding to the
+    passed in timestamp without taking any timezones into account, use croniter to
+    determine the next time that matches the cron string, translate that back into the passed in
+    timezone, and repeat, returning the first time that is later than the passed in timestamp.
+    """
+    start_datetime = pendulum.from_timestamp(start_timestamp, tz=timezone_str)
+
+    # Create a naive (timezone-free) datetime to pass into croniter
+    naive_time = datetime.datetime(
+        year=start_datetime.year,
+        month=start_datetime.month,
+        day=start_datetime.day,
+        hour=start_datetime.hour,
+        minute=start_datetime.minute,
+        second=start_datetime.second,
+        microsecond=start_datetime.microsecond,
+    )
+
+    # Go back an hour to ensure that we consider the full set of possible candidates (otherwise
+    # we might fail to properly consider a time that happens twice during a fall DST transition)
+    if ascending:
+        naive_time = naive_time - datetime.timedelta(hours=1)
+    else:
+        naive_time = naive_time + datetime.timedelta(hours=1)
```

Comment:
was starting at this for a bit to convince my self 1 hour was safe constant, some sort of comment that this value aligns with how (all I presume?) day light savings time transitions work might be helpful